dashboard/dataproviders/airflow.py

```diff
@@ -16,15 +16,29 @@ def get_dag_state(self, dag_id, execution_date):
         return result[0]['state']
 
     def get_history(self, days):
-        # used to skip the dag version while sorting in order to get only the most recent ones
-        extracted_dag_name_regex = '''SUBSTRING(dag_id FROM 1 FOR (REGEXP_INSTR(dag_id, '_v?[0-9]+.[0-9]+$') - 1))'''
-
         SQL = f'''
         SELECT dag_id, date, state FROM (
-            SELECT dag_id, execution_date as date, state, ROW_NUMBER() OVER (PARTITION BY {extracted_dag_name_regex} ORDER BY execution_date DESC) AS age
-            FROM dag_run
-        ) t WHERE age <= {days}
-        '''
+            SELECT
+                @row_number:=CASE
+                    WHEN @dag_id_current = clean_dag_id
+                      THEN
+                          @row_number + 1
+                      ELSE
+                           1
+                    END AS rownum,
+                @dag_id_current:=clean_dag_id clean_dag_id,
+                dag_id,
+                execution_date as date,
+                state
+            FROM (
+                SELECT *,
+                    CASE
+                        WHEN substring_index(reverse(dag_id), "v_", 1) rlike '[0-9]+\.[0-9]'
+                        THEN substring(dag_id, 1, length(dag_id) -1 - locate("v_", reverse(dag_id)))
+                        ELSE dag_id
+                    END as clean_dag_id from dag_run) dag_run, (SELECT @dag_id_current:='',@row_number:=0) as t
+            ORDER BY clean_dag_id, execution_date DESC
+        ) dr WHERE rownum <= {days}'''
```

Comment:
That looks much less readible to me than the previous version.
We'd have to comment the method like it was so it's clear it's done to ignore the DAG's specific version

Reply:
I agree, this is much less readable. But well, it's the only way to do it in one query for all DAGs on the DB that doesn't support windowing functions. It passes all the tests in the project.